### PR TITLE
Add initial cargo crate layout.

### LINKED ISSUE #1: Initial Project Layout
Hi,

as a first step, I think we need to figure out a rough project structure. Not that we can't change it later, but we need to come up with something.

Since I think OT Java is currently the most active developed/used (correct me if I'm wrong) it makes sense to take this as a starting point.

With that in mind, I propose the following:

 - `opentracing-rust` is a cargo workspace for the following subprojects.
 - `opentracing-api` provides only the traits which are required to implement a tracer but doesn't have any implementation by itself. It should be very lightweight and not make any assumptions on how the actual tracer implementation is done.
 - `opentracing-noop` a "noop" implementation of the API, can be used if no tracer needs to be used at runtime and also serves as a sanity check durning compilation.
 - `opentracing-mock` a simple mock implementation of the API, useful for testing and also showcases how to implement a simple tracer.

We might also need utils (which would hold the global tracer) and of course examples, but we can add those as soon as we need them.

ping @hanscj1 @stefano-pogliani @sile

### PR DESCRIPTION
**Pull request by daschl**
This change applies the basic repository layout discussed in

Fixes #1

### CODE REVIEW

**Comment by daschl** [comment]
@stefano-pogliani wdyt?

File: opentracing-api/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[package]
+name = "opentracing-api"
+version = "0.1.0"
+authors = ["The OpenTracing Developers"]
+license = "Apache-2.0"
+homepage = "http://opentracing.io"
```
Comment (stefano-pogliani) [inline opentracing-api/Cargo.toml]: Do we want to add the [documentation](https://doc.rust-lang.org/cargo/reference/manifest.html#the-documentation-field-optional) link from the start?
Docs say it is added for us but from what I've seen it was not.

Comment (stefano-pogliani) [reply opentracing-api/Cargo.toml]: Maybe add some `keywords` too?

Comment (stefano-pogliani) [reply opentracing-api/Cargo.toml]: Finally, is a maintain badge to ambitious?
```
# Maintenance: `status` is required Available options are `actively-developed`,
# `passively-maintained`, `as-is`, `none`, `experimental`, `looking-for-maintainer`
# and `deprecated`.
maintenance = { status = "actively-developed" }
```

**Review by stefano-pogliani — COMMENTED** [review]
Happy to add the extra attributes if we want them.

**Comment by daschl** [comment]
@stefano-pogliani added more attributes

**Review by stefano-pogliani — APPROVED** [review]
Looks great :+1:

**Comment by daschl** [comment]
Thanks!

**Pull request merged**